marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mapping/getDocument.sjs

```diff
@@ -0,0 +1,21 @@
+/**
+ Copyright (c) 2020 MarkLogic Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+const core = require('/data-hub/5/artifacts/core.sjs')
+
+xdmp.securityAssert("http://marklogic.com/data-hub/privileges/read-mapping", "execute");
+
+var stepName, uri;
+let mappingStep = core.getArtifact("mapping", stepName);
+fn.head(xdmp.eval("String(cts.doc('" + uri + "'))", null, {database: xdmp.database(mappingStep.sourceDatabase)}));
```

Comment:
Is it guaranteed that sourceDatabase will exist? I feel this needs to be a little more defensive - if sourceDatabase doesn't exist, I think it should just do a cts.doc(uri) on the current database (no eval).

Reply:
Also, if the mappingStep is not found, I think this should throw an error via ds-util - probably throwBadRequest.

Reply:
core.getArtifact throws HTTP 404 if step is not found